src/qopenglwebpage.cpp

```diff
@@ -186,6 +186,14 @@ void QOpenGLWebPage::setActive(bool active)
         mActive = active;
         d->mView->SetIsActive(mActive);
         Q_EMIT activeChanged();
+
+        disconnect(mozWindow(), &QMozWindow::drawOverlay,
+                   this, &QOpenGLWebPage::onDrawOverlay);
+
+        if (mActive) {
+            connect(mozWindow(), &QMozWindow::drawOverlay,
+                    this, &QOpenGLWebPage::onDrawOverlay, Qt::DirectConnection);
+        }
```

Comment:
The line 185 makes sure that this condition is checked only if mActive changes its value, so it's unlikely that `connect` is called twice when `setActive(true)` is called twice as well. But yes, for the sake of readability it's safer to try to disconnect always.